### DISCUSSION TITLE
Fixing Background Remove from Other Application

### DISCUSSION THREAD

**Discussion by MrPaulAR** [ImageMagick/ImageMagick#3800] [Help] [answered]
I've got a bunch of PNG files that I messed up the background removal (using a GUI) when I created the first layer then exported multiple images that used that background layer.

Is there a way to expand the transparent area into the center of the image by a few pixels or some other way to address this?  I'd much rather script this out than manually fix the background and re-export all the various layers.

Below is one of the sample images.  Each one is pretty much the exact same with different eyes & mouths.

![Arianna_ai_eo](https://user-images.githubusercontent.com/3765696/121071442-90e65200-c795-11eb-8fb2-e1c8dc39dd14.png)

**Comment by fmw42** [comment]
What platform? What version of Imagemagick.

Here is how to do that, but it will not work. Whatever tool you used has made the background black. You have lost your original background in this image.

We extract the alpha channel. Then dilate it, then put it back.

Unix syntax for ImageMagick 6. (If on ImageMagick 7, change `convert` to `magick`)

```
convert cartoon_face.png \
\( -clone 0 -alpha extract -morphology dilate disk:2 \) \
-alpha off -compose copy_opacity -composite \
cartoon_face_new.png

```

![cartoon_face_new](https://user-images.githubusercontent.com/13458218/121072330-ef0b3900-c785-11eb-875f-2737459743ee.png)


Input image without alpha:

![x](https://user-images.githubusercontent.com/13458218/121072372-fa5e6480-c785-11eb-93ca-01b52de8b171.png)


**If you have the original, then you can extract the alpha from this image, dilate it, then put it back into the original image before you did any background removal tracing.**

**Reply by MrPaulAR** [reply]
When I open the PNG in paint.net (tool I used to create the image) I see the transparent background.  Also, in chrome, the image I uploaded does show the github background instead of black so transparency seems to work.  Following is a screenshot from paint.net.

![image](https://user-images.githubusercontent.com/3765696/121073335-fdfae700-c797-11eb-84a3-d78ef43502dd.png)

Here is the raw file that I uploaded without copy & paste in case something is lost in translation.
![Arianna_e_eo](https://user-images.githubusercontent.com/3765696/121072934-7dd48180-c797-11eb-9794-6ea90428bc0d.png)

I'll be using v6 on linux.  

Looking at the image you created with the black background there is still the line around it.  Is this what you mean by saying "here is how you do it but it will not work"?

You've given me a start so I'll look into this and see if I can make it work.

**Reply by fmw42** [reply]
It is black under the transparency. Your original has a white background. I thought you had an image where there was color outside in the background that you wanted to recover or at least I thought you might have trimmed too much so that there was some of the color under the transparency.  I guess that is wrong.  What do you want to have filled in? Are you trying to extend whatever color you have left to make the hair larger? Please clarify further what you are trying to do.

**Comment by MrPaulAR** [comment]
My goal is to remove the colored outline around the head that's between it
and the transparent background.

Take a look at this zoomed in image.  You can see that ring around the
outside of the hair.

[image: image.png]

I would like it to be more like this [completely different] image where the
face model directly touches the black (or transparent...doesnt matter)
background.

[image: image.png]

**Comment by fmw42** [comment, marked as answer]
How about one of these:

convert cartoon_face.png \
\( -clone 0 -alpha extract -morphology erode diamond:1 \) \
-alpha off -compose copy_opacity -composite \
cartoon_face_new2.png

![cartoon_face_new2](https://user-images.githubusercontent.com/13458218/121076585-7018ff00-c78b-11eb-9ae3-4671390e65b1.png)


convert cartoon_face.png \
\( -clone 0 -alpha extract -morphology erode diamond:1 -blur 0x1 -level 50x100% \) \
-alpha off -compose copy_opacity -composite \
cartoon_face_new3.png

![cartoon_face_new3](https://user-images.githubusercontent.com/13458218/121076600-7313ef80-c78b-11eb-8177-d69e38e9ee37.png)

**Reply by MrPaulAR** [reply]
Perfect!  Thanks alot.
